src/blockchain/block/Position.cpp

```diff
@@ -29,7 +29,7 @@ Position::Position(const Height& height, const Hash& hash) noexcept
 }
 
 Position::Position(const Height& height, Hash&& hash) noexcept
-    : Position(Height{height}, std::move(hash))
+    : Position(std::move(Height{height}), std::move(hash))
```

Comment:
I don't get this change. For me `std::move(Height{height})` is r-value reference and `Height{height}` is also r-value reference and should gave the same output and invoke the same constructor.